packages/widget-chat/src/components/activity-item-post/index.js

```diff
@@ -2,22 +2,55 @@ import React, {PropTypes} from 'react';
 import classNames from 'classnames';
 
 import Avatar from '../avatar';
-import ActivityItemPostActions from '../activity-item-post-actions';
+import ActivityItemPostAction from '../activity-item-post-action';
 import styles from './styles.css';
 
+import {ICON_TYPE_DELETE, ICON_TYPE_FLAGGED_OUTLINE} from '../icon';
 
 export default function ActivityItemPost(props) {
+  function handleOnDelete() {
+    const {id, onActivityDelete} = props;
+    onActivityDelete(id);
+  }
+
+  function handleOnFlag() {
+    const {id, onActivityFlag} = props;
+    onActivityFlag(id);
+  }
+
+  function wrapActionItem(action, highlight) { // eslint-disable-line react/no-multi-comp
+    const actionClassNames = [`activity-post-action`, styles.activityPostAction];
+    if (highlight) {
+      actionClassNames.push(`activity-post-action-highlighted`, styles.activityPostActionHighlighted);
+    }
+    return (
+      <div className={classNames(actionClassNames)}>
+        {action}
+      </div>
+    );
+  }
+
   const {
     content,
-    id,
     isAdditional,
+    isFlagged,
     isSelf,
     name,
-    onActivityDelete,
     timestamp
   } = props;
 
-  const showDeleteAction = isSelf;
+  let deleteAction;
+  if (isSelf) {
+    deleteAction = <ActivityItemPostAction iconType={ICON_TYPE_DELETE} onClick={handleOnDelete} title="Delete this message" />;
+  }
+  else {
+    deleteAction = <div className={classNames(`action-spacer`, styles.actionSpacer)} />;
+  }
+  deleteAction = wrapActionItem(deleteAction, false);
+
+  let flagAction = <ActivityItemPostAction iconType={ICON_TYPE_FLAGGED_OUTLINE} onClick={handleOnFlag} title="Flag this message" />;
+  flagAction = wrapActionItem(flagAction, isFlagged);
+
```

Comment:
The wrapping of the action buttons seems overly complex. Instead of attaching the `isFlagged` highlighting class to just the button, would it be better to attach it to the `activity-post` similar to the `isAdditional` flag and just apply a style?